utils/templates/query.py.tpl

```diff
@@ -0,0 +1,352 @@
+
+    # Add type annotations for methods not defined in every subclass
+    __ror__: ClassVar[Callable[["Query", "Query"], "Query"]]
+    __radd__: ClassVar[Callable[["Query", "Query"], "Query"]]
+    __rand__: ClassVar[Callable[["Query", "Query"], "Query"]]
+
+    def __add__(self, other: "Query") -> "Query":
+        # make sure we give queries that know how to combine themselves
+        # preference
+        if hasattr(other, "__radd__"):
+            return other.__radd__(self)
+        return Bool(must=[self, other])
+
+    def __invert__(self) -> "Query":
+        return Bool(must_not=[self])
+
+    def __or__(self, other: "Query") -> "Query":
+        # make sure we give queries that know how to combine themselves
+        # preference
+        if hasattr(other, "__ror__"):
+            return other.__ror__(self)
+        return Bool(should=[self, other])
+
+    def __and__(self, other: "Query") -> "Query":
+        # make sure we give queries that know how to combine themselves
+        # preference
+        if hasattr(other, "__rand__"):
+            return other.__rand__(self)
+        return Bool(must=[self, other])
+
+
+{% for k in classes %}
+class {{ k.name }}({{ parent }}):
+    """
+    {% for line in k.docstring %}
+    {{ line }}
+    {% endfor %}
+    {% if k.args %}
+
+    {% for kwarg in k.args %}
+    {% for line in kwarg.doc %}
+    {{ line }}
+    {% endfor %}
+    {% endfor %}
+    {% endif %}
+    """
+    name = "{{ k.property_name }}"
+    {% if k.params %}
+    _param_defs = {
+        {% for param in k.params %}
+        "{{ param.name }}": {{ param.param }},
+        {% endfor %}
+        {% if k.name == "FunctionScore" %}
+        "filter": {"type": "query"},
+        "functions": {"type": "score_function", "multi": True},
+        {% endif %}
+    }
+    {% endif %}
+
+    def __init__(
+        self,
+        {% if k.args and not k.is_single_field and not k.is_multi_field %}
+        *,
+        {% endif %}
+        {% for arg in k.args %}
+        {{ arg.name }}: {{ arg.type }} = NOT_SET,
+        {% endfor %}
+        **kwargs: Any
+    ):
+        {% if k.name == "FunctionScore" %}
+        if isinstance(functions, NotSet):
+            functions = []
+            for name in ScoreFunction._classes:
+                if name in kwargs:
+                    functions.append({name: kwargs.pop(name)})  # type: ignore
+        {% elif k.is_single_field %}
+        if not isinstance(_field, NotSet):
+            kwargs[str(_field)] = _value
+        {% elif k.is_multi_field %}
+        if not isinstance(fields, NotSet):
+            for field, value in _fields.items():
+                kwargs[str(field)] = value
+        {% endif %}
+        super().__init__(
+            {% if not k.is_single_field and not k.is_multi_field %}
+            {% for arg in k.args %}
+            {{ arg.name }}={{ arg.name }},
+            {% endfor %}
+            {% endif %}
+            **kwargs
+        )
+
```

Comment:
There are a handful of Query sub-classes that had extra helper methods to make them more Pythonic. I'm just adding those here to keep things compatible with current code.